diff --git a/mobile/shared/actors/GeckoViewContentChild.sys.mjs b/mobile/shared/actors/GeckoViewContentChild.sys.mjs
index 9ca6e15bed3e..ee82063f248e 100644
--- a/mobile/shared/actors/GeckoViewContentChild.sys.mjs
+++ b/mobile/shared/actors/GeckoViewContentChild.sys.mjs
@@ -67,6 +67,37 @@ export class GeckoViewContentChild extends GeckoViewActorChild {
     return windowUtils.SCROLL_MODE_SMOOTH;
   }
 
+  collectSessionState() {
+    const { docShell, contentWindow } = this;
+    const history = lazy.SessionHistory.collect(docShell);
+    let formdata = SessionStoreUtils.collectFormData(contentWindow);
+    let scrolldata = SessionStoreUtils.collectScrollPosition(contentWindow);
+
+    // Save the current document resolution.
+    let zoom = 1;
+    const domWindowUtils = contentWindow.windowUtils;
+    zoom = domWindowUtils.getResolution();
+    scrolldata = scrolldata || {};
+    scrolldata.zoom = {};
+    scrolldata.zoom.resolution = zoom;
+
+    // Save some data that'll help in adjusting the zoom level
+    // when restoring in a different screen orientation.
+    const displaySize = {};
+    const width = {},
+      height = {};
+    domWindowUtils.getDocumentViewerSize(width, height);
+
+    displaySize.width = width.value;
+    displaySize.height = height.value;
+
+    scrolldata.zoom.displaySize = displaySize;
+
+    formdata = lazy.PrivacyFilter.filterFormData(formdata || {});
+
+    return { history, formdata, scrolldata };
+  }
+
   orientation() {
     const currentOrientationType = this.contentWindow?.screen.orientation.type;
     if (!currentOrientationType) {
